flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/debezium/DebeziumUtils.java

```diff
@@ -324,11 +324,22 @@ private static long getBinlogTimestamp(BinaryLogClient client, String binlogFile
             client.setBinlogFilename(binlogFile);
             client.setBinlogPosition(0);
 
-            LOG.info("begin parse binlog: {}", binlogFile);
+            LOG.info("Begin parse binlog: {}", binlogFile);
             client.connect();
         } finally {
             client.unregisterEventListener(eventListener);
         }
+        if (binlogTimestamps.isEmpty()) {
+            try {
+                if (client.isConnected()) {
+                    client.disconnect();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            LOG.warn("Failed to register eventListener and try to register it again");
+            return getBinlogTimestamp(client, binlogFile);
```

Comment:
Regardless of the exception, I think there is no need to retry when the queue is empty, the method should return and go to next binlog file.

Reply:
> Regardless of the exception, I think there is no need to retry when the queue is empty, the method should return and go to next binlog file.

if return and go to next binlog file. this binlog file will miss and lost data,beacause we don't kown this bninlog file's timestamp